src/canvas/PatternPaths/PatternPath.tsx

```diff
@@ -42,143 +33,40 @@ export abstract class PatternPath implements IPatternPath {
-            const points = path.getPoints();
-
-            const otherFirstPoint = points[0];
-            const otherLastPoint = points[points.length - 1];
-
-            if(!updatedFirstPoint && myFirstPoint.isWithinRadius(otherFirstPoint, radius)) {
-                this.snapStartToPoint(otherFirstPoint);
-                updatedFirstPoint = true;
-            }
-
-            if(!updatedFirstPoint && myFirstPoint.isWithinRadius(otherLastPoint, radius)) {
-                this.snapStartToPoint(otherLastPoint);
-                updatedFirstPoint = true;
-            }
-
-            if(!updatedLastPoint && myLastPoint.isWithinRadius(otherFirstPoint, radius)) {
-                this.snapEndToPoint(otherFirstPoint);
-                updatedLastPoint = true;
-            }
-
-            if(!updatedLastPoint && myLastPoint.isWithinRadius(otherLastPoint, radius)) {
-                this.snapEndToPoint(otherLastPoint);
-                updatedLastPoint = true;
-            }
-
-        }
-
-        if (updatedFirstPoint || updatedLastPoint) {
-            this._updatePath2D();
-        }
-    };
-
     getLengthInPixels = (): number => {
-        if (!this._fittedSegment) {
-            throw new Error();
-        }
-        return this._fittedSegment.getLength();
+        return this._segment.getLength();
     };
 
     getSegment = (): Segment => {
-        if (!this._fittedSegment) {
-            throw new Error();
-        }
-        return this._fittedSegment;
+        return this._segment;
     };
 
-    setFittedSegment = (): void => {
-        this._setFittedSegment();
-        this._points = this._fittedSegment ? this._fittedSegment.computePoints() : new Array<Point>();
-    };
-
-    splitAtPoint = (interesection: Point): PatternPath[] => {
+    splitAtPoint = (intersection: Point): PatternPath[] => {
         const originalSegment = this.getSegment();
-        const splitSegments = originalSegment.split(interesection);
+        const splitSegments = originalSegment.split(intersection);
         if (splitSegments.length !== 2) {
             throw new Error("Split did not return correct number of segments");
         }
 
         return this._createPathsFromSegments(splitSegments);
     };
 
-    protected abstract _createPathsFromSegments(segments: Segment[]): PatternPath[];
-    protected abstract _setFittedSegment(): void;
-    protected abstract _updatePath2D(): void;
-    protected abstract _addPoint(point: Point): void;
+    protected _createPathsFromSegments = (segments: Segment[]): PatternPath[] => {
+        const paths: PatternPath[] = [];
+        segments.forEach(segment => {
+            paths.push(new PatternPath(this._type, segment));
+        });
+
+        return paths;
+    };   
+    
+    protected _updatePath2D = (): void => {
+        this._path2D = new Path2D();
+        const start = this._segment.getStart();
+        this._path2D.moveTo(start.x, start.y);
+        this._segment.drawTo(this._path2D);
```

Comment:
Should drawTo just be draw here?

Reply:
I named it drawTo because the method does not do the whole job of drawing: it does not do the moveTo part. Previously the draw method did the whole job. 

This will make more sense when we have more segments in one patternPath: after the moveTo, we will loop through the segments and call drawTo on each one, so that one patternPath will appear as one continuous stroke